backendImpl/src/main/java/com/virtuslab/gitmachete/backend/impl/GitMacheteRepository.java

```diff
@@ -604,16 +606,16 @@ private SyncToRemoteStatus deriveSyncToRemoteStatus(IGitCoreLocalBranchSnapshot
       return syncToRemoteStatus;
     }
 
-    private boolean hasJustBeenCreated(IGitCoreLocalBranchSnapshot branch) throws GitCoreException {
+    private boolean hasJustBeenCreated(IGitCoreLocalBranchSnapshot branch) {
       List<IGitCoreReflogEntry> reflog = deriveFilteredReflog(branch);
       return reflog.isEmpty() || reflog.head().getOldCommitHash().isEmpty();
     }
 
-    private SyncToParentStatus deriveSyncToParentStatus(
+    protected SyncToParentStatus deriveSyncToParentStatus(
         IGitCoreLocalBranchSnapshot coreLocalBranch,
         IGitCoreLocalBranchSnapshot parentCoreLocalBranch,
-        @Nullable GitMacheteForkPointCommit forkPoint)
-        throws GitCoreException {
+        @Nullable GitMacheteForkPointCommit forkPoint) throws GitCoreException {
```

Comment:
`GitMacheteException` at this level?

Reply:
And in a few other places within this file... :thinking: Ofc we can agree that these exceptions constraints are the most important for API

Reply:
Yeah, I'm aware of that... I was using `GitMacheteException`s everywhere initially, but this required rewrapping `GitCoreException`s in too many places, leading to less readable code :/ so I decided to wrap them on the higher level, closer to the API boundary